src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -358,15 +358,19 @@ public async Task<IActionResult> SetDeploymentTarget(string sessionId, [FromBody
 
             var serviceProvider = CreateSessionServiceProvider(state);
             var orchestrator = CreateOrchestrator(state, serviceProvider);
+            var cloudApplicationNameGenerator = serviceProvider.GetRequiredService<ICloudApplicationNameGenerator>();
 
-            if(!string.IsNullOrEmpty(input.NewDeploymentRecipeId) &&
-               !string.IsNullOrEmpty(input.NewDeploymentName))
+            if (!string.IsNullOrEmpty(input.NewDeploymentRecipeId) && !string.IsNullOrEmpty(input.NewDeploymentName))
```

Comment:
What do we do in the push image case where name is not required? I worry we are getting lucky that the toolkit is accidently sending us a name that we ignore.

Reply:
Added logic to exclude the stack name validation in the push image case